$OpenBSD: patch-pwd_shadow_c,v 1.3 2014/10/30 12:20:30 jasper Exp $

Add new sp_loginclass field to retrieve the user's login class.

--- pwd/shadow.c.orig	Thu Oct 30 11:16:01 2014
+++ pwd/shadow.c	Thu Oct 30 11:26:02 2014
@@ -66,6 +66,7 @@ static VALUE convert_pw_struct( struct passwd *entry )
          INT2FIX(difftime(entry->pw_change, 0) / (24*60*60)), /* pw_change */
          INT2FIX(difftime(entry->pw_expire, 0) / (24*60*60)), /* sp_expire */
          Qnil, /* sp_flag */
+         rb_tainted_str_new2(entry->pw_class), /* sp_loginclass, user access class */
          NULL);
 }
 
@@ -107,7 +108,8 @@ Init_shadow()
                                      "sp_namp","sp_pwdp","sp_lstchg",
                                      "sp_min","sp_max","sp_warn",
                                      "sp_inact","pw_change",
-                                     "sp_expire","sp_flag", NULL);
+                                     "sp_expire","sp_flag",
+				     "sp_loginclass", NULL);
   rb_sGroupEntry = rb_struct_define("GroupEntry",
                                     "sg_name","sg_passwd",
                                     "sg_adm","sg_mem",NULL);
